README.md
Jed Hammoud

<!-- README.md is generated from README.Rmd. Please edit that file -->

GITHUB ID jh5764a

# HW04Jh5764a

<!-- badges: start -->
<!-- badges: end -->

The goal of HW04Jh5764a is to gain information about data frames.
Function 1 allows you to get the names of each variable in a data set,
allows you to get the class of each variable, allows you to find the
number of unique data points in each variable and allows you to find the
number of NA’s in each variable. Function 2 Allows you to visualize the
distribution and density of the data in each variable in a given data
frame. It works by taking a specific class as an input and then returns
a histogram or bar plot of the all of the variables of the specific
class.

## Installation

You can install the development version of HW04Jh5764a like so:

This package is only available by permission of the author on my github
repo using the install_github() function from the (devtools) package.

What is special about using `README.Rmd` instead of just `README.md`?
You can include R chunks like so:

``` r
summary(cars)
#>      speed           dist       
#>  Min.   : 4.0   Min.   :  2.00  
#>  1st Qu.:12.0   1st Qu.: 26.00  
#>  Median :15.0   Median : 36.00  
#>  Mean   :15.4   Mean   : 42.98  
#>  3rd Qu.:19.0   3rd Qu.: 56.00  
#>  Max.   :25.0   Max.   :120.00
```

You’ll still need to render `README.Rmd` regularly, to keep `README.md`
up-to-date. `devtools::build_readme()` is handy for this. You could also
use GitHub Actions to re-render `README.Rmd` every time you push. An
example workflow can be found here:
<https://github.com/r-lib/actions/tree/v1/examples>.

You can also embed plots, for example:

<img src="man/figures/README-pressure-1.png" width="100%" />

In that case, don’t forget to commit and push the resulting figure
files, so they display on GitHub.


NEWS.md
# HW04Jh5764a 0.0.0.9000

* Added a `NEWS.md` file to track changes to the package.


R/function1.R

#' @name Function-1
#' @title Function 1 creates a new data frame
#'
#' @param dataset
#'
#'
#'
#' @return returns a data frame containing data like names of columns number of unique and NA values and the class of each variable
#' @export

#'
#' @examples
#' A <- c("J","e","d","H","a","m","m","o","u","d")
#' B <- c(1,2,3,4,1,2,3,4,4,6)
#' C <- c(1,2,3,2,1,2,1,2,3,3)
#' D <- c("a","b","c","c","d","d","e","f","g","f")
#' E <- c(1,4,5,6,7,4,5,42,8,11)
#' f <-c(TRUE,TRUE,FALSE,FALSE,FALSE,TRUE,TRUE,FALSE,FALSE,TRUE)
#' Example1 <- data.frame(A,B,C,D,E,f)
#' A1 <- c("g","f","d","d","l","n","z","w","ud","d")
#' B2 <- c(11,22,33,44,11,22,33,44,44,66)
#' C3 <- c(1,4,9,4,1,4,1,8,33,1089)
#' D4 <- c("a","b","c","c","d","d","e","f","g","f")
#' E5 <- c(FALSE,FALSE,FALSE,TRUE,TRUE,TRUE,FALSE,FALSE,TRUE,TRUE)
#' f6 <-c(TRUE,TRUE,FALSE,TRUE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE)
#' Example2 <- data.frame(A1,B2,C3,D4,E5,f6)
#' examp1 <- Function_1(Example1)
#' examp1
#' examp2 <- Function_1(Example2)
#' examp2






Function_1 <- function(dataset){
  column_names <- c(colnames(dataset))
  class_data <- c(sapply(dataset,class))
  unique_data <- c(sapply(dataset, function(x) NROW(unique(x))))
  na_data <- c(sapply(dataset, function(y) sum(length(which(is.na(y))))))
  data.frame(column_names,class_data,unique_data,na_data)
}






R/function2.R


#' Function 2 creates plots based on the class input in the function
#'
#' @param dataset is a data frame of the users choosing
#' @param class_name
#'
#' @return returns a histogram of the variables that are of class class_name if the class is numeric, double, or integer, returns a bar plot of the variables that are of class class_name if the class is factor, logical, or character
#' @export
#'
#' @examples
#' A <- c("J","e","d","H","a","m","m","o","u","d")
#' B <- c(1,2,3,4,1,2,3,4,4,6)
#' C <- c(1,2,3,2,1,2,1,2,3,3)
#' D <- c("a","b","c","c","d","d","e","f","g","f")
#' E <- c(1,4,5,6,7,4,5,42,8,11)
#' f <-c(TRUE,TRUE,FALSE,FALSE,FALSE,TRUE,TRUE,FALSE,FALSE,TRUE)
#' Example1 <- data.frame(A,B,C,D,E,f)
#' A1 <- c("g","f","d","d","l","n","z","w","ud","d")
#' B2 <- c(11,22,33,44,11,22,33,44,44,66)
#' C3 <- c(1,4,9,4,1,4,1,8,33,1089)
#' D4 <- c("a","b","c","c","d","d","e","f","g","f")
#' E5 <- c(FALSE,FALSE,FALSE,TRUE,TRUE,TRUE,FALSE,FALSE,TRUE,TRUE)
#' f6 <-c(TRUE,TRUE,FALSE,TRUE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE)
#' Example2 <- data.frame(A1,B2,C3,D4,E5,f6)
#' examp1A <- Function_2(Example1, "numeric")
#' examp1A
#' examp2A <- Function_2(Example2, "logical")
#' examp2A







Function_2 <- function(dataset, class_name){
  class_data <- c(sapply(dataset,class))

  if(!(class_name %in% class_data))
    stop("the class name you've chosen does not exist in this dataset")

  if (class_name == "numeric"){
    dataset1 <- dplyr::select(dataset, tidyselect::vars_select_helpers$where(is.numeric))

    if (length(dataset1 > 0)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_histogram()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)->Graph1

    }

    else {
      print("there are no class numerics in this dataset.")
    }
  }

  else if (class_name == "double"){
    dataset1 <- dplyr::select(dataset, tidyselect::vars_select_helpers$where(is.double))

    if (length(dataset1 > 0)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_histogram()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)->Graph1

    }

    else {
      print("there are no class double in this dataset.")
    }
  }
  else if (class_name == "integer"){
    dataset1 <- dplyr::select(dataset, tidyselect::vars_select_helpers$where(is.integer))

    if (length(dataset1 > 0)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_histogram()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)->Graph1

    }

    else {
      print("there are no class integer in this dataset.")
    }
  }
  else if (class_name == "character"){
    dataset1 <- dplyr::select(dataset, tidyselect::vars_select_helpers$where(is.character))

    if (length(dataset1 > 0)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_bar()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)->Graph1

    }
    else if (length(dataset1 > 7)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_bar()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)+
        ggplot2::coord_flip()->Graph1

    }
    else {
      print("there are no class character in this dataset.")
    }
  }
  else if (class_name == "factor"){
    dataset1 <- dplyr::select(dataset, tidyselect::vars_select_helpers$where(is.factor))

    if (length(dataset1 <= 7) && length(dataset1 > 0)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_bar()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)->Graph1

    }
    else if (length(dataset1 > 7)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_bar()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)+
        ggplot2::coord_flip()->Graph1

    }
    else {
      print("there are no class factor in this dataset.")
    }
  }
  else if (class_name == "logical"){
    dataset1 <- dplyr::select(dataset, tidyselect::vars_select_helpers$where(is.logical))

    if (length(dataset1 <= 7) && length(dataset1 > 0)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_bar()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)->Graph1

    }
    else if (length(dataset1 > 7)) {
      dataset1 %>%
        tidyr::pivot_longer(cols = c(1:length(dataset1)),names_to = "var", values_to = "values") -> dataset1

      ggplot2::ggplot(dataset1, ggplot2::aes(x = values))+
        ggplot2::geom_bar()+
        ggplot2::ggtitle(paste0("Variables in dataset of class ", class_name))+
        ggplot2::facet_wrap(var ~.)+
        ggplot2::coord_flip()->Graph1

    }
    else {
      print("there are no class logical in this dataset.")
    }

  }
  Graph1
}


